java-frontend/src/main/java/org/sonar/java/se/MethodBehavior.java

```diff
@@ -19,16 +19,71 @@
  */
 package org.sonar.java.se;
 
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+
+import org.sonar.java.resolve.JavaSymbol;
+import org.sonar.java.se.symbolicvalues.SymbolicValue;
 import org.sonar.plugins.java.api.semantic.Symbol;
+import org.sonar.plugins.java.api.tree.VariableTree;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
 public class MethodBehavior {
   private final Symbol.MethodSymbol methodSymbol;
+  private final List<MethodYield> yields;
+  private final Map<Symbol, SymbolicValue> parameters;
 
   public MethodBehavior(Symbol.MethodSymbol methodSymbol) {
     this.methodSymbol = methodSymbol;
+    this.yields = new ArrayList<>();
+    this.parameters = new LinkedHashMap<>();
   }
 
+  public void createYield(ProgramState programState) {
+    List<Symbol> parameterSymbols = methodSymbol.declaration().parameters().stream().map(VariableTree::symbol).collect(Collectors.toList());
+    MethodYield.ConstrainedSymbolicValue[] parametersCSV = new MethodYield.ConstrainedSymbolicValue[parameterSymbols.size()];
+
+    int i = 0;
+    for (Symbol parameterSymbol : parameterSymbols) {
```

Comment:
This should probalby be done functionally through a stream and collected to an array.